solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceDirectInputOperator.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.solace;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+import com.solacesystems.jcsmp.BytesXMLMessage;
+import com.solacesystems.jcsmp.Consumer;
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.Topic;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.InputOperator;
+import com.datatorrent.api.Operator;
+import com.datatorrent.netlet.util.DTThrowable;
+
+public abstract class AbstractSolaceDirectInputOperator<T> extends AbstractSolaceBaseInputOperator<T> implements InputOperator, Operator.ActivationListener<com.datatorrent.api.Context.OperatorContext>
+{
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractSolaceDirectInputOperator.class);
+
+  private final transient AtomicReference<Throwable> throwable;
+  private transient Context.OperatorContext context;
+  protected transient long currentWindowId;
+
+  @NotNull
+  protected String topicName;
+  private transient Topic topic;
+  private transient BytesXMLMessage recentMessage = null;
+
+  private transient List<T> messages = new ArrayList<T>();
+  protected final transient Map<Long, T> currentWindowRecoveryState;
+  protected static final transient int DEFAULT_BUFFER_SIZE = 500;
```

Comment:
transient not necessary for a static final integer constant. @PramodSSImmaneni ?